File: ssms/installation-management/src/main/java/com/ssms/installationmanagement/entity/Installation.java
package com.ssms.installationmanagement.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "installations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Installation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "installation_id")
    private Long id;

    @NotBlank(message = "Job reference is required")
    @Column(name = "job_reference", nullable = false, length = 30, unique = true)
    private String jobReference;

    @NotNull(message = "Order ID is required")
    @Column(name = "order_id", nullable = false)
    private Long orderId;

    @NotNull(message = "Customer ID is required")
    @Column(name = "customer_id", nullable = false)
    private Long customerId;

    @Column(name = "technician_id")
    private Long technicianId;

    @NotNull(message = "Technician is required")
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "technician_id", nullable = false, insertable = false, updatable = false)
    private Technician technician;

    @Column(name = "scheduled_by_user_id")
    private Long scheduledByUserId;

    @NotNull(message = "Scheduled date is required")
    @Column(name = "scheduled_date", nullable = false)
    private LocalDateTime scheduledDate;

    @Column(name = "completed_date")
    private LocalDateTime completedDate;

    @NotBlank(message = "Installation address is required")
    @Column(name = "installation_address", nullable = false, columnDefinition = "TEXT")
    private String installationAddress;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private InstallationStatus status = InstallationStatus.SCHEDULED;

    @Column(name = "technician_notes", columnDefinition = "TEXT")
    private String technicianNotes;

    @Column(name = "cancellation_reason", columnDefinition = "TEXT")
    private String cancellationReason;

    @Column(name = "is_deleted", nullable = false)
    private Boolean isDeleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/installation.ts
// ====================================
// INSTALLATION MANAGEMENT TYPES
// Created by: [Your Name]
// Purpose: Type definitions for installation management system
// ====================================

export enum InstallationStatus {
  SCHEDULED = 'SCHEDULED',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED'
}

export enum AvailabilityStatus {
  AVAILABLE = 'AVAILABLE',
  OFF = 'OFF',
  ON_JOB = 'ON_JOB'
}

export interface Installation {
  id: number;
  jobReference: string;
  orderId: number;
  customerId: number;
  technicianId: number;
  scheduledByUserId?: number;
  scheduledDate: string;
  completedDate?: string;
  installationAddress: string;
  status: InstallationStatus;
  technicianNotes?: string;
  cancellationReason?: string;
  isDeleted: boolean;
  deletedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface Technician {
  id: number;
  userId: number;
  specialization: string;
  certificationNumber: string;
  availabilityStatus: AvailabilityStatus;
  phone?: string;
  hiredDate: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateInstallationRequest {
  jobReference: string;
  orderId: number;
  customerId: number;
  technicianId: number;
  scheduledDate: string;
  installationAddress: string;
  technicianNotes?: string;
}

export interface UpdateInstallationRequest {
  jobReference?: string;
  orderId?: number;
  customerId?: number;
  technicianId?: number;
  scheduledDate?: string;
  installationAddress?: string;
  technicianNotes?: string;
}

export interface StatusUpdateRequest {
  status: InstallationStatus;
  technicianNotes?: string;
  cancellationReason?: string;
}

export interface TechnicianAssignmentRequest {
  technicianId: number;
}